nsxt/resource_nsxt_policy_host_switch_profile.go

```diff
@@ -0,0 +1,592 @@
+								ValidateFunc: validation.StringInSlice(lacpTimeoutTypes, false),
+							},
+							"uplinks": {
+								Type:        schema.TypeSet,
+								Description: "uplink names",
+								Required:    false,
+								Optional:    false,
+								Computed:    true,
+								Elem: &schema.Resource{
+									Schema: map[string]*schema.Schema{
+										"uplink_name": getDataSourceRequiredStringSchema("Name of this uplink"),
+										"uplink_type": {
+											Type:         schema.TypeString,
+											Description:  "Type of the uplink",
+											Required:     true,
+											ValidateFunc: validation.StringInSlice(uplinkTypes, false),
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				"teaming": {
+					Type:        schema.TypeSet,
+					Description: "Default TeamingPolicy associated with this UplinkProfile",
+					Required:    true,
+					MaxItems:    1,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"active_list": {
+								Type:        schema.TypeSet,
+								Description: "List of Uplinks used in active list",
+								Required:    true,
+								Elem: &schema.Resource{
+									Schema: map[string]*schema.Schema{
+										"uplink_name": getDataSourceRequiredStringSchema("Name of this uplink"),
+										"uplink_type": {
+											Type:         schema.TypeString,
+											Description:  "Type of the uplink",
+											Required:     true,
+											ValidateFunc: validation.StringInSlice(uplinkTypes, false),
+										},
+									},
+								},
+							},
+							"policy": {
+								Type:         schema.TypeString,
+								Description:  "Teaming policy",
+								Required:     true,
+								ValidateFunc: validation.StringInSlice(teamingPolicies, false),
+							},
+							"rolling_order": {
+								Type:        schema.TypeBool,
+								Description: "Flag for preemptive mode",
+								Default:     false,
+								Optional:    true,
+							},
+							"standby_list": {
+								Type:        schema.TypeSet,
+								Description: "List of Uplinks used in standby list",
+								Optional:    true,
+								Elem: &schema.Resource{
+									Schema: map[string]*schema.Schema{
+										"uplink_name": getDataSourceRequiredStringSchema("Name of this uplink"),
+										"uplink_type": {
+											Type:         schema.TypeString,
+											Description:  "Type of the uplink",
+											Required:     true,
+											ValidateFunc: validation.StringInSlice(uplinkTypes, false),
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				"named_teamings": {
+					Type:        schema.TypeSet,
+					Description: "List of named uplink teaming policies that can be used by logical switches",
+					Optional:    true,
+					MaxItems:    32,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"name": getDataSourceRequiredStringSchema("The name of the uplink teaming policy"),
```

Comment:
Is this schema any differerent for `teaming` schema above?

Reply:
They are both "teaming" at the end of the day. But `teaming` is just for the default teaming which is required and the `named_teaming` is optional we could provide more teamings to this profile. To summarize the differences:
`teaming`:
1. It is a required field.
2. It can only have one item, the default teaming.
3. We can't config the name of this default teaming.

`named_teaming`: 
1. It is an optional field
2. We could provide up to 32 items.
3. We could config the name for each item.